pkg/util/shortid_generator.go

```diff
@@ -26,6 +26,16 @@ func IsShortUIDTooLong(uid string) bool {
 }
 
 // GenerateShortUID generates a short unique identifier.
+// This will return a valid k8s name
 func GenerateShortUID() string {
-	return shortid.MustGenerate()
+	uid, err := uuid.NewRandom()
+	if err != nil {
+		panic("invalid uuid")
+	}
+	uuid := uid.String()
+	return string(alphaRunes[uidrand.Intn(len(alphaRunes))]) + // alpha
+		uuid[0:8] + // time_low
+		uuid[9:13] + // time_mid
+		uuid[19:22] + // Clock sequence + variant
+		uuid[28:] // Node
```

Comment:
Not sure whether we gain much by stripping out the dashes beyond making the result slightly more compact.  If we're going to do that we may as well just generate our own random alphanumeric string since that's effectively what we're producing indirectly (because we're generating random uuids).

The comments here aren't actually relevant to random uuids since the majority of the uuid is random (except the 7th and 9 bits which encode the version and variant).

We could do the same de-hyphenation with `strings.ReplaceAll(uid.String(), "-", "")`, but I think we should either treat these as uuids and not strip the hyphens or just generate our own random strings.

As far as the first character, the uuid itself is random and the first byte has no special meaning so we can just replace the first character with a random alphabetic character instead of prepending one if we want to produce standard-formatted uuids.

With that all said, I don't think that this kind of random generated uid is a very friendly option for an identifier if it's going to be used as a k8s name.

Reply:
I updated this to just use standard UUID and replace the first character.

> I don't think that this kind of random generated uid is a very friendly option for an identifier if it's going to be used as a k8s name.

maybe, but that is a different question.  Regardless what we do with k8s, it still seems weird to assign random UIDs that do not adhere to any widely used standard.